### ISSUE TITLE
Problem building dockerfile-with-features after upgrading to debian bookworm

### ISSUE THREAD

**Issue by emilhvid** [microsoft/vscode-remote-release#8202]
Type: <b>Bug</b>

When every i try to build the dev container it throws the following error:
```
Dockerfile-with-features:1
--------------------
   1 | >>> # syntax=docker/dockerfile:1.4
   2 |     ARG _DEV_CONTAINERS_BASE_IMAGE=placeholder
   3 |     ARG VARIANT=3-bullseye
--------------------
ERROR: failed to solve: unlazy requires an applier
Error: Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-ehh/container-features/0.32.0-1678872163723/Dockerfile-with-features -t vsc-dev_ops_tools-45aedc8c9b62a50d591dd74f03fab4b1 --target dev_containers_target_stage --build-arg VARIANT=3.10-bullseye --build-arg NODE_VERSION=lts/* --build-context dev_containers_feature_content_source=/tmp/devcontainercli-ehh/container-features/0.32.0-1678872163723 --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /home/ehh/repos/dev_ops_tools
    at pie (/home/ehh/.vscode/extensions/ms-vscode-remote.remote-containers-0.284.0/dist/spec-node/devContainersSpecCLI.js:1921:1698)
    at async DF (/home/ehh/.vscode/extensions/ms-vscode-remote.remote-containers-0.284.0/dist/spec-node/devContainersSpecCLI.js:1920:3917)
    at async k7 (/home/ehh/.vscode/extensions/ms-vscode-remote.remote-containers-0.284.0/dist/spec-node/devContainersSpecCLI.js:1920:2836)
    at async Bie (/home/ehh/.vscode/extensions/ms-vscode-remote.remote-containers-0.284.0/dist/spec-node/devContainersSpecCLI.js:1935:2631)
    at async Wf (/home/ehh/.vscode/extensions/ms-vscode-remote.remote-containers-0.284.0/dist/spec-node/devContainersSpecCLI.js:1935:3756)
    at async ooe (/home/ehh/.vscode/extensions/ms-vscode-remote.remote-containers-0.284.0/dist/spec-node/devContainersSpecCLI.js:2059:17326)
    at async soe (/home/ehh/.vscode/extensions/ms-vscode-remote.remote-containers-0.284.0/dist/spec-node/devContainersSpecCLI.js:2059:17067)
{"outcome":"error","message":"Command failed: docker buildx build --load --build-arg BUILDKIT_INLINE_CACHE=1 -f /tmp/devcontainercli-ehh/container-features/0.32.0-1678872163723/Dockerfile-with-features -t vsc-dev_ops_tools-45aedc8c9b62a50d591dd74f03fab4b1 --target dev_containers_target_stage --build-arg VARIANT=3.10-bullseye --build-arg NODE_VERSION=lts/* --build-context dev_containers_feature_content_source=/tmp/devcontainercli-ehh/container-features/0.32.0-1678872163723 --build-arg _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label --build-arg _DEV_CONTAINERS_IMAGE_USER=root --build-arg _DEV_CONTAINERS_FEATURE_CONTENT_SOURCE=dev_container_feature_content_temp /home/ehh/repos/dev_ops_tools","description":"An error occurred building the image."}
```
However this seems to only be a problem in #syntax=docker/dockerfile:1.4.0
see: https://github.com/docker/buildx/issues/1481#issuecomment-1359575756

If i  edit the Dockerfile-with-features accordingly and execute the command i can build the container image just fine.

I am running with 

docker version:
Client: Docker Engine - Community
 Version:           23.0.1
 API version:       1.42
 Go version:        go1.19.5
 Git commit:        a5ee5b1
 Built:             Thu Feb  9 19:46:51 2023
 OS/Arch:           linux/amd64
 Context:           default

docker buildx version: github.com/docker/buildx v0.10.2 00ed17d
devcontainer version: 0.31.0

vscode: 1.76.1
5e805b79fcb6ba4c2d23712967df89a089da575b
x64

VS Code version: Code 1.76.1 (5e805b79fcb6ba4c2d23712967df89a089da575b, 2023-03-08T16:31:31.010Z)
OS version: Linux x64 6.1.0-6-amd64
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|12th Gen Intel(R) Core(TM) i9-12900 (24 x 3848)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 1, 0|
|Memory (System)|125.65GB (118.18GB free)|
|Process Argv|--unity-launch --crash-reporter-id 8ac323a8-392a-4e79-bbf4-6203b3faf98d|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|lightdm-xsession|
|XDG_CURRENT_DESKTOP|X-Cinnamon|
|XDG_SESSION_DESKTOP|lightdm-xsession|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (30)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-mermaid|bie|1.17.7
mermaid-markdown-syntax-highlighting|bpr|1.5.1
dotnet-test-explorer|for|0.7.8
copilot|Git|1.77.9225
markdown-toc|jof|1.4.0
restructuredtext-pack|lex|1.0.1
azure-pipelines|ms-|1.208.0
vscode-docker|ms-|1.24.0
csharp|ms-|1.25.4
isort|ms-|2022.8.0
python|ms-|2023.4.1
vscode-pylance|ms-|2023.3.20
jupyter|ms-|2023.2.1200692131
jupyter-keymap|ms-|1.1.0
jupyter-renderers|ms-|1.0.15
vscode-jupyter-cell-tags|ms-|0.1.6
vscode-jupyter-slideshow|ms-|0.1.5
remote-containers|ms-|0.282.0
remote-ssh|ms-|0.98.0
remote-ssh-edit|ms-|0.84.0
remote-wsl|ms-|0.76.1
vscode-remote-extensionpack|ms-|0.24.0
azure-account|ms-|0.11.3
cmake-tools|ms-|1.13.45
cpptools|ms-|1.14.4
cpptools-extension-pack|ms-|1.3.0
remote-explorer|ms-|0.2.0
remote-server|ms-|1.0.0
jinjahtml|sam|0.20.0
cmake|twx|0.0.17

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vslsvsres303:30308271
pythonvspyl392:30443607
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscoreces:30445986
pythondataviewer:30285071
vscod805cf:30301675
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdf:30486550
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30681850
azure-dev_surveyone:30548225
vscccc:30610679
pyindex848:30662994
nodejswelcome1:30587005
2e4cg342:30602488
pyind779:30671433
89544117:30613380
pythonsymbol12:30671437
a9j8j154:30646983
vsctsb:30677850
pythonb192:30669360
funwalk2cf:30682975
pythonms35:30686772

```

</details>

We have written the needed data into your clipboard because it was too large to send. Please paste.

**Comment by esemeniuc** [comment]
I solved this by killing all docker instances, and restarting docker (`sudo systemctl stop docker.service`), then removing all images: `docker system prune -a --volumes`, and restarting docker service

**Comment by emilhvid** [comment]
> I solved this by killing all docker instances, and restarting docker (`sudo systemctl stop docker.service`), then removing all images: `docker system prune -a --volumes`, and restarting docker service

Unfortunatly pruning my docker did not resolve the issue.

> [docker/buildx#1481 (comment)](https://github.com/docker/buildx/issues/1481#issuecomment-1359575756) seems to suggest that using 1.4.0 or later would fix the issue? Since we are using 1.4, shouldn't that be 1.4.0 or later?
> 
> What was the edit in the Dockerfile that made it work?

If i remove the applier entierly  or use the exact version `# syntax=docker/dockerfile:1.4.0` i am able to build. As i read the issue, buildx will fail, unless an exact version is given for the applier. As the example given `# syntax=docker/dockerfile:1` would also be evaluated to something newer than 1.4.0 but fails to resolve just as i experience with `# syntax=docker/dockerfile:1.4`.
